### DISCUSSION TITLE
How do we want to move forward regarding callbacks

### DISCUSSION THREAD

**Discussion by primeapple** [okuuva/auto-save.nvim#17] [General]
Hey @okuuva , what do you think of the current implementation of the `callbacks`?

- Is there really any value to have callbacks that run before and after enabling of the Plugin?
- Is there really any value to have callbacks after the save?
- Is there really any value to have callbacks before asserting the save (in fact I already removed it here: https://github.com/okuuva/auto-save.nvim/pull/16 )?

The only meaningful callback seems to be the `before_saving` callback, since it can do things like formatting or linting.
We could probably achieve the same by implementing our own Autocommand event, right? That would be a lot cleaner than having to squash all the callback logic into the config of this plugin.

Do you use callbacks anywhere?
One usecase would be for debugging, but I would rather have #13 implemented.

**Comment by okuuva** [comment]
Personally I have no use for any of them. And I agree about `before_saving` being replaceable with an Autocommand event. Actually, another session-management plugin I use (`persisted.nvim`) [made the switch from callbacks to Autocommands a while ago](https://github.com/olimorris/persisted.nvim/issues/51) and while the initial change in the config was a bit of a PITA the end result is much cleaner. So +1 for axing them in favor of our own Autocommand events.

**Reply by primeapple** [reply]
That sounds very good. Will have a look into them. We should be able to remove all the callbacks except for `before_saving` for now.

**Comment by primeapple** [comment]
Implemented in https://github.com/okuuva/auto-save.nvim/pull/25
